### PR TITLE
Builtin Resources new constexpr storage data updates

### PR DESCRIPTION
**Pull request by AnastaZIuk**

### CODE REVIEW

**Comment by AnastaZIuk** [comment]
there was also an attempt of making per built-in resource data compilations I want to save https://github.com/Devsh-Graphics-Programming/Nabla/commit/d5fd97a9539708cbe2da905f3db2d25c621349a3, but decided to not go into this because it needs us to generate another PCH/reuse & even with that it's just slow and not worth it

File: include/nbl/core/xxHash256.h [hunk 1]
```diff
@@ -132,7 +133,92 @@ inline void XXHash_256(const void* input, size_t len, uint64_t* out)
 	out[3] += v4;
 }
 
+constexpr std::array<uint64_t, 4> XXHash_256(const uint8_t* input, const size_t len)
+{
```
Comment (devshgraphicsprogramming) [inline include/nbl/core/xxHash256.h:137]: 1. missing namesapce
2. implement the non-constexpr in terms of this one

File: src/nbl/builtin/builtinHeaderGen.py [hunk 2]
```diff
@@ -1,94 +1,119 @@
...
+        std::tm modified = {{ // Absolute time since 1970 in UTC+0
+            .tm_sec = 6,
+            .tm_min = 9,
+            .tm_hour = 6,
+            .tm_mday = 9,
+            .tm_mon = 6,
+            .tm_year = 9,
+            .tm_isdst = 0
+        }};
+    }};
```
Comment (devshgraphicsprogramming) [inline src/nbl/builtin/builtinHeaderGen.py:83]: no reason for every archive namespace to re-define this struct, slap it in `nbl::system` somewhere

File: include/nbl/core/xxHash256.h (hunk 1 above)
Comment (AnastaZIuk) [reply include/nbl/core/xxHash256.h:137]: resolved in following commit

File: src/nbl/builtin/builtinHeaderGen.py (hunk 2 above)
Comment (AnastaZIuk) [reply src/nbl/builtin/builtinHeaderGen.py]: resolved

**Pull request merged**